src/Diagnostics.DataProviders/DataProviderConfigurations/KustoDataProviderConfiguration.cs

```diff
@@ -186,6 +186,22 @@ public string KustoApiEndpoint
         [ConfigurationName("Retry:ExceptionsToRetryFor")]
         public string ExceptionsToRetryFor { get; set; }
 
+        /// <summary>
+        /// List of , separated clusters which requests will be shadowed from
+        /// e.g. "wawswusfollower,wawseusfollower"
+        /// </summary>
+        [ConfigurationName("QueryShadowingClusterMaping:ClusterNameGroupings")]
+        public string QueryShadowingClusterNameGroupings { get; set; }
+
+        /// <summary>
+        /// List of , separated cluster lists seprated by : which requests will be shadowed to
+        /// e.g. "wustestcluster1:wustestcluster2,eustestcluster1"
+        /// </summary>
+        [ConfigurationName("QueryShadowingClusterMaping:TestingClusterNamesGroupings")]
+        public string QueryShadowingTestClustersGroupings { get; set; }
```

Comment:
Would it work if only QueryShadowingTestClustersGroupings setting existed and QueryShadowingClusterNameGroupings did not?

Reply:
No, these strings are used in `PostInitialize` below, and only both of them are not null or empty, then the `QueryShadowingClusterMapping` dictionary will be initialized